### PR TITLE
OOP Adapters: Add support for abstract and concrete value types in instances

### PR DESCRIPTION
**Pull request by mandarjog**
1. Add support for abstract ValueType, and specific types like IPAddress, TimeStamp, and Duration.
2. Refactor code for better sharing.
3. Add named handlers to static encoder

### CODE REVIEW

**Comment by mandarjog** [comment]
partially addresses #7645

File: bin/mixer_codegen.sh [hunk 1]
```diff
@@ -98,6 +100,8 @@ if [ "$opttemplate" = true ]; then
     "google/protobuf/any.proto:github.com/gogo/protobuf/types"
     "gogoproto/gogo.proto:github.com/gogo/protobuf/gogoproto"
     "google/protobuf/duration.proto:github.com/gogo/protobuf/types"
+    "google/protobuf/timestamp.proto=github.com/gogo/protobuf/types"
+    "google/rpc/status.proto=github.com/gogo/googleapis/google/rpc"
```
Comment (douglas-reid) [inline bin/mixer_codegen.sh]: Did you mean to add `struct.proto` here too, or is that not needed in templates?

Comment (mandarjog) [reply bin/mixer_codegen.sh]: done.

File: mixer/pkg/protobuf/yaml/dynamic/valueTypeEncoder.go [hunk 2]
```diff
@@ -50,135 +52,7 @@ func valueTypeEncoderBuilder(_ *descriptor.DescriptorProto, fd *descriptor.Field
 		if isConstString {
 			vVal.Value = &v1beta1.Value_StringValue{sval}
 		} else {
```
Comment (douglas-reid) [inline mixer/pkg/protobuf/yaml/dynamic/valueTypeEncoder.go]: super nit: you could avoid the `else` and indent with something like:

```
if isConstString {
  vVal.Value = &v1beta1.Value_StringValue{sval}
  break
}
return buildExprEncoder(sval, fd.GetTypeName(), compiler)
```

**Pull request merged**